### PR TITLE
IGNITE-21273 Document newly-added thread pools in README files

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-21273

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/network/README.md [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+nodes. ScaleCube also sends its messages over this transport.
+
+## Channel types
+A couple of nodes might have a channel per each channel type. Each of these channels is only used to exchange messages
+using the given channel type. There a few standard channel types:
+
+ * Default (used by default when channel type is not specified), id=0
+ * Scalecube (used by all ScaleCube messages), id=1
+ * Deploy (used to send deployment artifacts that might be quite heavy), id=2
+ * Files (used to exchange with files), id=3
```
Comment (sashapolo) [inline modules/network/README.md:14]:
```suggestion
 * Files (used to exchange files), id=3
```

File: modules/network/README.md [hunk 2]
```diff
@@ -38,6 +48,12 @@ Message handling can also be offloaded to another thread:
 Note that in this case the network message would be considered **handled** before it is processed  
 by another thread.
 
+The default `MessagingService` implementation handles incoming messages in the inbound thread pool, threads are named
+`{consistentId}-MessagingService-inbound-X-Y` (one per channel type), where X stands per channel ID. Deserialization of the
+incoming messages happens in these threads as well.
+
+It also uses an outbound thread `{consistentId}-MessagingService-outbound-X` for sending messages (in this thread serialization happens).
```
Comment (sashapolo) [inline modules/network/README.md:55]:
```suggestion
It also uses an outbound thread `{consistentId}-MessagingService-outbound-X` for sending messages (message serialization also happens in this thread).
```

File: modules/runner/README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 This document describes user-level and component-level cluster lifecycles and their mutual interaction.
 
 ## Node lifecycle
-A node maintains its' local state in the local persistent key-value storage named vault. The data stored in the vault is 
+A node maintains its local state in the local persistent key-value storage named vault. The data stored in the vault is
```
Comment (sashapolo) [inline modules/runner/README.md:5]: Could you also please remove the part about the metastorage projected properties?

File: modules/runner/README.md [hunk 4]
```diff
@@ -203,3 +203,10 @@ Besides local node stopping logic two more actions took place on a cluster as a
  * Both range and watch cursors will be removed on server side. Given process is linearized with meta storage
   operations by using a meta storage raft.
  * Baseline update and corresponding baseline recalculation with ongoing partition raft groups redeployment.
+
+## Threading
```
Comment (sashapolo) [inline modules/runner/README.md:207]: This parts looks like out of nowhere, are you sure this is the correct place or that it even needs to be mentioned?

File: modules/runner/README.md (hunk 3 above)
Comment (rpuch) [reply modules/runner/README.md:5]: It seems that there are more parts that need to be updated, I filed an issue: https://issues.apache.org/jira/browse/IGNITE-21294

File: modules/runner/README.md (hunk 4 above)
Comment (rpuch) [reply modules/runner/README.md]: The section actually looks alien here, I moved it to the `threading.md`

**Review by sashapolo — APPROVED** [review]

**Pull request merged**